data_structures/binary_search_tree.cpp

```diff
@@ -75,6 +76,10 @@ int findMaxInLeftST(node* n) {
  * \param[in] x the int value of a node
  */
 void Remove(std::shared_ptr<node>* p, std::shared_ptr<node>* n, int x) {
+    if (*n == nullptr) {
```

Comment:
What you had originally was also fine.
you can pass `p` and `n` and `const std:shared_ptr<node>` (this is the appropriate way) and still do the 

```cpp
if (n == nullptr) return;
```

Reply:
yes, I try to used const reference as you suggested , but const reference dosen't work well here.
We need to modify the "node n" reference value  in function Remove().
So , the system suggest  to use pointer.